src/pipecat/services/heygen/api.py

```diff
@@ -0,0 +1,242 @@
+#
+# Copyright (c) 2024–2025, Daily
+#
+# SPDX-License-Identifier: BSD 2-Clause License
+#
+
+"""HeyGen API.
+
+API to communicate with HeyGen Streaming API.
+
+"""
+
+from enum import Enum
+from typing import Any, Dict, Literal, Optional
+
+import aiohttp
+from loguru import logger
+from pydantic import BaseModel, Field
+
+
+class AvatarQuality(str, Enum):
+    """Enum representing different avatar quality levels."""
+
+    low = "low"
+    medium = "medium"
+    high = "high"
+
+
+class VoiceEmotion(str, Enum):
+    """Enum representing different voice emotion types."""
+
+    EXCITED = "excited"
+    SERIOUS = "serious"
+    FRIENDLY = "friendly"
+    SOOTHING = "soothing"
+    BROADCASTER = "broadcaster"
+
+
+class ElevenLabsSettings(BaseModel):
+    """Settings for ElevenLabs voice configuration."""
+
+    stability: Optional[float] = None
+    similarity_boost: Optional[float] = None
+    style: Optional[int] = None
+    use_speaker_boost: Optional[bool] = None
+
+
+class VoiceSettings(BaseModel):
+    """Voice configuration settings."""
+
+    voice_id: Optional[str] = Field(None, alias="voiceId")
+    rate: Optional[float] = None
+    emotion: Optional[VoiceEmotion] = None
+    elevenlabs_settings: Optional[ElevenLabsSettings] = Field(None, alias="elevenlabsSettings")
+
+
+class NewSessionRequest(BaseModel):
+    """Requesting model for creating a new HeyGen session."""
+
+    avatarName: str
+    quality: Optional[AvatarQuality] = None
+    knowledgeId: Optional[str] = None
+    knowledgeBase: Optional[str] = None
+    voice: Optional[VoiceSettings] = None
+    language: Optional[str] = None
+    version: Literal["v2"] = "v2"
+    video_encoding: Literal["H264"] = "H264"
+    source: Literal["sdk"] = "sdk"
+    disableIdleTimeout: Optional[bool] = None
+
+
+class HeyGenSession(BaseModel):
+    """Response model for a HeyGen session."""
+
+    session_id: str
+    access_token: str
+    realtime_endpoint: str
+    url: str
+
+
+class HeygenApiError(Exception):
+    """Custom exception for HeyGen API errors."""
+
+    def __init__(self, message: str, status: int, response_text: str) -> None:
+        """Initialize the HeyGen API error.
+
+        Args:
+        message: Error message
```

Comment:
Indent `Args` by 4 spaces.

```suggestion
            message: Error message
```